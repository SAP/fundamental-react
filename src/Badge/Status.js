import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { STATUS_TYPES } from '../utils/constants';

const Status = ({ type, glyph, children, className, ...props }) => {
    const statusClasses = classnames(
        'fd-status-label',
        {
            [`fd-status-label--${type}`]: !!type,
            [`sap-icon--${glyph}`]: !!glyph
        },
        className
    );

    return (
        <span
            {...props}
            className={statusClasses}>
            {children}
        </span>
    );
};
Status.displayName = 'Status';

Status.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    glyph: PropTypes.string,
    type: PropTypes.oneOf(STATUS_TYPES)
};

export default Status;
