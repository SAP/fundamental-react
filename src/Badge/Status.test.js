import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Status from './Status';

describe('<Status />', () => {

    const defaultStatus = <Status>Default</Status>;
    const typeStatus = (
        <Status className='blue' type='success'>
            Default
        </Status>
    );
    const iconStatus = <Status glyph='history'>Default</Status>;

    test('create status', () => {
        // create default status
        let component = renderer.create(defaultStatus);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create success type status
        component = renderer.create(typeStatus);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // create icon status
        component = renderer.create(iconStatus);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the Status component', () => {
            const element = mount(<Status data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <Status ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('SPAN');
        expect(ref.current.className).toEqual('fd-status-label');
    });
});
