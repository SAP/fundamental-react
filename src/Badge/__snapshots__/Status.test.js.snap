// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> create status 1`] = `
<span
  className="fd-status-label"
>
  Default
</span>
`;

exports[`<Status /> create status 2`] = `
<span
  className="fd-status-label fd-status-label--success blue"
>
  Default
</span>
`;

exports[`<Status /> create status 3`] = `
<span
  className="fd-status-label sap-icon--history"
>
  Default
</span>
`;
